### PR TITLE
[RTI-16181] Add files reload functionality

### PR DESCRIPTION
**Pull request by pbrudnick**

### CODE REVIEW

File: src/geodata2.erl [hunk 1]
```diff
@@ -82,15 +82,13 @@ new(ConfigName, Ets) ->
 
 -spec init(_) -> {ok, state()} | {stop, tuple()}.
 init(_Args) ->
-    case new(dbfile, ?GEODATA2_STATE_TID) of
+    Res = load_files(),
+    maybe_launch_reload(),
```
Comment (raz-adroll) [inline src/geodata2.erl:86]: 🤔 The name is a little misleading since it will not launch anything, I think a better name would be `maybe_schedule_reload`

**Review by raz-adroll — COMMENTED** [review]
Can we add a test for reload?

File: src/geodata2.erl [hunk 2]
```diff
@@ -125,3 +130,22 @@ id(X) ->
...
+            error_logger:error_report({error, geodata2_cannot_load_dbfile}),
+            Error
+    end.
+
+maybe_launch_reload() ->
+    case get_env(geodata2, reload_milliseconds) of
+        {ok, Time} when is_integer(Time) ->
+            erlang:send_after(Time, self(), reload);
+        _ ->
+            undefined
```
Comment (pehuen-rodriguez) [inline src/geodata2.erl:150]: Perhaps `noop` for not schedulling is a little more expressive

**Review by raz-adroll — APPROVED** [review]

**Comment by pbrudnick** [comment]
Boodah test deploy

test deploy looks good: 
https://app.datadoghq.com/dashboard/4bm-ca6-4kn/rtb-diagnostic-board?fromUser=true&refresh_mode=sliding&tpl_var_control%5B0%5D=r16186-ctrl&tpl_var_target%5B0%5D=r16186&view=spans&from_ts=1723669412112&to_ts=1723673012112&live=true

ETS with peak after 1h due to duplicate-rename-delete operation just 300Mb:
https://app.datadoghq.com/metric/explorer?fromUser=true&start=1723669642166&end=1723672774196&paused=true#N4Ig7glgJg5gpgFxALlAGwIYE8D2BXJVEADxQEYAaELcqyKBAC1pEbghkcLIF8qo4AMwgA7CAgg4RKUAiwAHOChASAtnADOcAE4RNIKtrgBHPJoQaUAbVBGN8qVoD6gnNtUZCKiOq279VKY6epbINiAiGOrKQdpYZAYgUJ4YThr42gDGSsgg6gi6mZaBZnHKHsTIGlgaCHCqAHTqjXhaUMAAbjoakiLI2mQAbGQAHIMAVDwABABGWFOd3b08IHygkdG5sVgATInJCKnpeFk5eYiFxSDbyhgdMFU1dY3NDa1w7V3aPVL9Q6MTaZzBZfH4iFZrCJRM7bADM+xSaQy2XKFwgRUSN1ydweOkwIhgTXqbiwDUQGkW316f2GY0ms3mlLBEIAulRXO48JhQuEOaouRgYqV4qsKKA+QKhTpdqLxW5+ZgpXF4Tw2SBalg0DlQPIMFqEHVlFAcDAnJluT1Mok0KI4E45IplOkbVBrbanPQmMoRPK9aKkhB7JgsPaFGcbSIlKqeHx1fIbQgAMJSYQwFAiLloHhAA

**Review by pehuen-rodriguez — APPROVED** [review]

**Comment by pbrudnick** [comment]
Test deployed in Dyno:
https://app.datadoghq.com/dashboard/nev-xcw-vmw/dyno-release?fromUser=false&refresh_mode=paused&tpl_var_exchange%5B0%5D=staging&view=spans&from_ts=1723732375694&to_ts=1723734875235&live=false

**Pull request merged**